### DISCUSSION TITLE
Often see "xdrip is not responding"

### DISCUSSION THREAD

**Discussion by zdooder** [NightscoutFoundation/xDrip#3544] [Bug reports]
I've been using xDrip+ for several years, since I was on a meter and MDI (!! for it's awesome predictability engine), to using a CGM (Dexcom G6, then G7), to using a pump, etc.  Since I moved over to the Dexcom G7, and there was no BYOD app to broadcast to xDrip+ (I was not satisfied with the lack of backfilled data for xDrip+), I opted to abandon the Dexcom G7 app and use the native G7 support in xDrip+.  But since doing this, I very often see "xDrip is not responding" popups occurring.  I generally just hit "Wait" and the notice goes away, and there don't see to be any ill effects, but I feel like this is not great and I'd like to try to troubleshoot it.

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/9edb27ce-b0e0-43bb-823c-65bbf4afcdb7)

The Hardware Data Source "page" just gives me:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/20e6b0e3-43a5-4935-84d1-8ae90e8e63dd)

But the Dex Debug Settings are:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/69cd4364-9963-4375-a6d0-28c7b4d4198a)
![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/e08a16d1-3725-4276-8bfa-c945feff81a4)

My Classic Status Page is:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/5262a7f5-ac47-479e-ae50-4eadce0e2576)

The G5/G6/G7 Status Page is:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/17d043b0-c133-4b7c-87fc-ec8a8dd2f2e6)

The phone is a Samsung Galaxy Fold 4, running Android 14.  Phone Hardware Information:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/60bb711a-77d6-4f59-bcfa-bbde0e22a37e)

I do have a Galaxy Watch5, if that matters, for which I have enabled "Android Wear Integration" with the following settings:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/19bb36b3-4235-4b5d-babf-1b8cae74c0ec)

Finally, if I need to (or just should) downgrade, I'm happy to do that.  I figured it best to try the latest available version to rule out the possibility that an older version was causing my problem.  The latest nightly seems to be no different, however (but also, no worse).

Any suggestions?

**Comment by Navid200** [comment]
Thanks for the screenshots.

xDrip not responding is a catastrophic failure.
I am surprised you are seeing that.

@jamorham

**Reply by jimbaxter** [reply]
I see this error about once a day for a few months. Though  I have a different setup of OOP2 and the Freestyle Libre 2.
It usually goes away immediately once you hit wait.

Although I have found that recently it is sounding the alarm for missed BG readings after 30 minutes a quite often which it never used to do, this is when I have the phone on me. I thought it might be a Pixel 8 Bluetooth hardware issue, though perhaps it relates to the not responding issue?

**Reply by Navid200** [reply]
From the 3-dot menu button at the top right corner of the main screen, go to `Import Export features` &#8722;> `Cloud Backup` &#8722;> `Select Backup location`.  
Log in to your Google Account to enable Google Drive backup.  

Now, `Do backup now`.  
Then, you should see something like this:  
![Screenshot_20240706-101557](https://github.com/NightscoutFoundation/xDrip/assets/51497406/9ed43fb5-e709-4ddc-a246-ef63aa92f40b)
<br/>  
  
What does it show as the backup size?

**Comment by PhilTU** [comment]
Does your watch collect directly from the sensor without the phone ? If not you can remove the "Enable wear collection service" option, which may be irrelevant to your problem but worth a try.

**Reply by zdooder** [reply]
I am not certain if the watch collects directly from the G7 or not (but I suspect it has based on the "Watch Service State: Service Stopped <timestamp>" message on the status screen).  I thought it would be useful for the watch to collect data when the phone was out of range, but it's not a requirement (very rarely is the phone out of range for any serious length of time, and the only real benefit is not having to pull out the pump to check).

I tried disabling the "Enable Wear Collection Service" option, but immediately got the "not responding" message, and have gotten it additional times since then with the option still disabled.

**Reply by PhilTU** [reply]
> but immediately got the "not responding" message,

ok, was worth a try as it won't collect direct to the watch AFAIK no was something that could be eliminated.

**Comment by zdooder** [comment]
I didn't think to include in my initial write-up, but I have a NightScout service set up on a VPS.  It's not used for much, but I do have a few things that query it for recent BG levels to display them.  The xDrip+ configuration for this is:

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/d8f944b1-b5d4-4b9a-a3e0-9427fcf1e6fe)

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/9cd0c447-fb68-490c-a752-990506be0dee)

![image](https://github.com/NightscoutFoundation/xDrip/assets/2055522/010bbf71-b80a-4570-b1d2-981286f87da4)

I got to wondering if the uploads (or downloads) from that service might be a factor.

I also noticed when checking this that I also had enabled "Dexcom Share Server Upload", probably a long time ago when I was concerned about my provider wanting to use Dexcom Clarity to monitor my treatments.  I know (now) that the two ("share" and "clarity") are not related, but I didn't then.  Additionally, now that I'm on a pump, they use t:connect to perform this monitoring, as it shows CGM readings in addition to treatments.  I have disabled the "Dexcom Share Server Upload" function to see if that might be the cause of the problem.

**Comment by jamorham** [comment]
Most likely thing I would suspect would be too much data stored. What `glucose retention` setting do you have? I would suggest setting it to 365 to only store a year's worth of data.  If its not that we can try to dig deeper.

**Comment by zdooder** [comment]
I had the glucose retention set to unlimited.  I changed it to 365 days.  I did get a few more of the not responding messages, but it hasn't occurred in some time, so I believe that this ultimately was the cause and it's resolved now.  Thanks so much for your help!
